test/util/discovery/discovery.go

```diff
@@ -41,7 +41,7 @@ func DiscoveredResources(clients *client.ClientSet, sriovInfos *cluster.EnabledN
 			if err != nil {
 				continue
 			}
-			device, ok := filterDevices(node.ObjectMeta.Name, sriovDeviceList)
+			device, ok := filterDevices(policy, node.ObjectMeta.Name, sriovDeviceList)
```

Comment:
if the goal is to find the device associated with the resourcename, I think a better way is to pass via the nodestate spec:

```
  spec:
    dpConfigVersion: "1607381"
    interfaces:
    - name: ens1f0
      numVfs: 5
      pciAddress: "0000:12:00.0"
      vfGroups:
      - deviceType: netdevice
        policyName: test-sctppolicyq7xbg
        resourceName: sctptestres
        vfRange: 0-4
```

Here you don't depend on the selector of the policy, but you have the information you need to match the resource with the device.

Reply:
This test does not really need finding the specific nic. Removing this check for now. This will reappear in a following PR (using the nodestate spec)